src/dropout.jl

```diff
@@ -62,6 +63,7 @@ dropout!(B::AbstractArray, A::AbstractArray, p::Real; dims = :) = dropout!(_rng_
 function dropout!(rng::AbstractRNG, dst::AbstractArray, src::AbstractArray, p::Real; dims=:)
     size(dst) == size(src) || throw(DimensionMismatch("dropout! expects output array the same size as input"))
     0 <= p <= 1 || throw(ArgumentError("dropout expects a probability 0 <= p <= 1"))
+    _rng_compat_array(rng, A)
```

Comment:
```suggestion
    _rng_compat_array(rng, src)
```
It also occurs to me that the docstring is not in line with the actual function signature. Any reason to use `A`/`B` instead of `dst`/`src` for the former?

Reply:
No real reason, just ended up this way.